src/providers/ipa/ipa_selinux.c

```diff
@@ -762,67 +749,19 @@ static void selinux_child_step(struct tevent_req *subreq)
     close(state->io->write_to_child_fd);
```

Comment:
Hi,

I know this is already an issue in the old code, but shouldn't `close()` be called before the `return` in case of an error to avoid a leaking file descriptor?

bye,
Sumit

Reply:
IIUC, `child_io_destructor()` should take care of this:
https://github.com/SSSD/sssd/blob/85784e7614da54aeaed2e8a757b5bd3c9523125d/src/providers/ipa/ipa_selinux.c#L600

The reason this explicit close() is needed here: so that 'selinux_child' "knows" that there are no more pending data.

Reply:
Hi,

thanks, you are right, I forgot to search for a destructor.

bye,
Sumit